examples/MenuExample.swift

```diff
@@ -0,0 +1,102 @@
+// This example demonstrates the minimal path to building a custom transition using the Material
+// Motion Transitioning APIs in Swift. The essential steps have been documented below.
+
+class MenuExampleViewController: ExampleViewController {
+
+  func didTap() {
+    let modalViewController = ModalViewController()
+
+    // The transition controller is an associated object on all UIViewController instances that
+    // allows you to customize the way the view controller is presented. The primary API on the
+    // controller that you'll make use of is the `transition` property. Setting this property will
+    // dictate how the view controller is presented. For this example we've built a custom
+    // MenuTransition, so we'll make use of that now:
+    modalViewController.transitionController.transition = MenuTransition()
+
+    // Note that once we assign the transition object to the view controller, the transition will
+    // govern all subsequent presentations and dismissals of that view controller instance. If we
+    // want to use a different transition (e.g. to use an edge-swipe-to-dismiss transition) then we
+    // can simply change the transition object before initiating the transition.
+
+    present(modalViewController, animated: true)
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    let label = UILabel(frame: view.bounds)
+    label.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+    label.textColor = .white
+    label.textAlignment = .center
+    label.text = "Tap to start the transition"
+    view.addSubview(label)
+
+    let tap = UITapGestureRecognizer(target: self, action: #selector(didTap))
+    view.addGestureRecognizer(tap)
+  }
+
+  override func exampleInformation() -> ExampleInfo {
+    return .init(title: type(of: self).catalogBreadcrumbs().last!,
+                 instructions: "Tap to present a modal transition.")
+  }
+}
+
+// Transitions must be NSObject types that conform to the Transition protocol.
+private final class MenuTransition: NSObject, Transition {
+
+  // The sole method we're expected to implement, start is invoked each time the view controller is
+  // presented or dismissed.
+  func start(with context: TransitionContext) {
+    let fromVC = context.backViewController
+    let toVC = context.foreViewController
+    let containerView = context.containerView
+    
+    if(context.direction == .forward) {
+      containerView.insertSubview(toVC.view, aboveSubview: fromVC.view)
+      toVC.view.frame.origin.x = -1 * toVC.view.frame.width
+      UIView.animate(
+        withDuration: 0.6,
+        animations: {
+          toVC.view.frame.origin.x = -1 * (toVC.view.frame.width / 2)
+        },
+        completion: { _ in
+          context.transitionDidEnd()
+        }
+      )
+      if let snapshot = fromVC.view.snapshotView(afterScreenUpdates: false) {
+        snapshot.isUserInteractionEnabled = false
+        containerView.insertSubview(snapshot, belowSubview: toVC.view)
```

Comment:
It looks like you're trying to preserve the back view controller's view after the transition completes.

- [ ] You can use `TransitionWithPresentation` to achieve this effect instead by returning `.overCurrentContext` in `defaultModalPresentationStyle`.
- [ ] You can then delete the snapshotting code from the transition.